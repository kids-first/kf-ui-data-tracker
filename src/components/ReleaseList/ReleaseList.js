import React from 'react';
import {Header, Loader, Comment, Icon, Image, Table} from 'semantic-ui-react';
import {Link} from 'react-router-dom';
import Markdown from 'react-markdown';
import {longDate} from '../../common/dateUtils';

const ReleaseList = ({loading, releases}) => {
  if (loading) {
    return <Loader>Loading...</Loader>;
  }

  if (!loading && releases.length === 0) {
    return (
      <Header disabled textAlign="center" as="h4">
        No releases data available for current study.
      </Header>
    );
  }

  return (
    <Comment.Group size="large">
      {releases.map(r => (
        <Comment key={r.node.id}>
          <Comment.Avatar as={Icon} name="tag" />
          <Comment.Content>
            <Comment.Author className="display-inline">
              <Link
                className="text-blue"
                to={'/releases/history/' + r.node.kfId}
              >
                {r.node.version + ' - ' + r.node.name + ' '}
              </Link>
            </Comment.Author>
            <Comment.Metadata className="text-12">
              on {longDate(r.node.createdAt)}
            </Comment.Metadata>
            <Comment.Text>
              <Markdown
                source={r.node.description}
                renderers={{
                  image: Image,
                  table: props => <Table>{props.children}</Table>,
                }}
                linkTarget="_blank"
              />
            </Comment.Text>
          </Comment.Content>
        </Comment>
      ))}
    </Comment.Group>
  );
};

export default ReleaseList;
