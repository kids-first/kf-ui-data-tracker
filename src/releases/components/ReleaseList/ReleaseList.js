import React from 'react';
import {Header, Loader, List, Icon, Image, Table} from 'semantic-ui-react';
import Markdown from 'react-markdown';
import {longDate} from '../../../common/dateUtils';

const ReleaseList = ({loading, releases}) => {
  const coordUrl = process.env.REACT_APP_COORD_UI + 'releases/';

  if (loading) {
    return <Loader>Loading...</Loader>;
  }

  if (!loading && releases.length === 0) {
    return (
      <Header disabled textAlign="center" as="h4">
        No releases data available for current study.
      </Header>
    );
  }

  return (
    <List divided>
      {releases.map(r => (
        <List.Item>
          <List.Icon name="tag" verticalAlign="middle" />
          <List.Content>
            <List.Header>
              <a
                className="text-blue"
                as="a"
                href={`${coordUrl + r.node.kfId}`}
                rel="noopener noreferrer"
                target="_blank"
              >
                {r.node.version + ' - ' + r.node.name + ' '}
              </a>
              <Icon link size="small" name="external" />
            </List.Header>
            <List.Description>on {longDate(r.node.createdAt)}</List.Description>
            <List.Description>
              <Markdown
                source={r.node.description}
                renderers={{
                  image: Image,
                  table: props => <Table>{props.children}</Table>,
                }}
              />
            </List.Description>
          </List.Content>
        </List.Item>
      ))}
    </List>
  );
};

export default ReleaseList;
