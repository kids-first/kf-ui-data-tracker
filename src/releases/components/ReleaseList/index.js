export {default as ReleaseList} from './ReleaseList';
