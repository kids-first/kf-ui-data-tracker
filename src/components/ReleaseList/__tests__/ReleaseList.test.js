import React from 'react';
import {render, cleanup} from '@testing-library/react';
import getStudyReleases from '../../../../__mocks__/kf-api-release-coordinator/responses/getStudyReleases.json';
import ReleaseList from '../ReleaseList';

afterEach(cleanup);
it('renders ReleaseList on loading stage', async () => {
  const {container} = render(<ReleaseList loading releases={[]} />);
  expect(container).toMatchSnapshot();
});

it('renders ReleaseList on empty stage', async () => {
  const {container} = render(<ReleaseList releases={[]} />);
  expect(container).toMatchSnapshot();
});

it('renders ReleaseList on normal stage', async () => {
  const releases = getStudyReleases.data.study.releases.edges;
  const {container} = render(<ReleaseList releases={releases} />);
  expect(container).toMatchSnapshot();
});
