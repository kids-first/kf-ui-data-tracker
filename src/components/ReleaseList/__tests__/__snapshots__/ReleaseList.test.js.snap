// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ReleaseList on empty stage 1`] = `
<div>
  <h4
    class="ui disabled center aligned header"
  >
    No releases data available for current study.
  </h4>
</div>
`;

exports[`renders ReleaseList on loading stage 1`] = `
<div>
  <div
    class="ui text loader"
  >
    Loading...
  </div>
</div>
`;

exports[`renders ReleaseList on normal stage 1`] = `
<div>
  <div
    class="ui large comments"
  >
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_FXR1JBZA"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.64.0 - Test ETL 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          5 Nov 2019
        </div>
        <div
          class="text"
        >
          <h1>
            Header 1
          </h1>
          <ol>
            <li>
              numbered list
            </li>
            <li>
              and more
            </li>
          </ol>
          <p>
            <img
              class="ui image"
              src="https://imgc.artprintimages.com/img/print/white-bunny-rabbit-wearing-tiara-close-up-studio-shot_u-l-q10d2j30.jpg"
            />
          </p>
          <table
            class="ui table"
          >
            <thead>
              <tr>
                <th>
                  Tables
                </th>
                <th
                  style="text-align: center;"
                >
                  Are
                </th>
                <th
                  style="text-align: right;"
                >
                  Cool
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  col 3 is
                </td>
                <td
                  style="text-align: center;"
                >
                  right-aligned
                </td>
                <td
                  style="text-align: right;"
                >
                  $1600
                </td>
              </tr>
              <tr>
                <td>
                  col 2 is
                </td>
                <td
                  style="text-align: center;"
                >
                  centered
                </td>
                <td
                  style="text-align: right;"
                >
                  $12
                </td>
              </tr>
              <tr>
                <td>
                  zebra stripes
                </td>
                <td
                  style="text-align: center;"
                >
                  are neat
                </td>
                <td
                  style="text-align: right;"
                >
                  $1
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_HPFK6H9Z"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.58.0 - Fix diagnosis 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          17 Sept 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_X1X0MCSY"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.55.0 - Biospecimens into diagnosis 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          4 Sept 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_M40RKAKH"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.53.0 - Fix outcome 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          6 Aug 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_B20Z2QR3"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.52.0 - Fix outcome 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          1 Aug 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_YD0E9JQT"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.51.0 - Phenotype nested 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          30 Jul 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_VENNYKVE"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.50.0 - Revert phenotype 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          25 Jul 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_V6KY1M2F"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.48.0 - Fix family composition 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          23 Jul 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_4EP89WT7"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.47.0 - Fix family members 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          22 Jul 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_XQ3XARN3"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.46.0 - Phenotypes nested 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          17 Jul 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_01JTCGXM"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.41.0 - Test ETL 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          28 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_42XS0PHM"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.40.0 - Test ETL 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          28 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_9R6CJCQR"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.39.0 - Test ETl 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          24 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_SQB0BNM0"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.36.0 - Test aws policy 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          17 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_CZ9C1V4G"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.35.0 - Test Rollcall 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          15 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_6D1C4KQD"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.34.0 - Test ETL 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          15 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_BNFXBYPM"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.33.0 - Test ETL 8 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          15 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_NPKBGKEQ"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.32.0 - Test ETL 7 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          15 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_7XJPEGDZ"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.31.0 - issue-15461 2 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          2 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_W384YPKK"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.30.0 - issue-66 test 3 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          2 May 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_MDC6Y3S8"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.29.0 - issue-15461 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          15 Apr 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_BDM6JA10"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.28.0 - dcf externalid-latestdid test 2 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          10 Apr 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_AKFCSQMZ"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.27.0 - dcf externalid-latestdid test 1 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          10 Apr 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_3H0P5GY0"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.25.0 - repo fix again 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          26 Mar 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_VQNNVGD5"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.23.0 - fix repos try 3 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          26 Mar 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_FA9A6QGK"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.22.0 - redeploy test 2 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          26 Mar 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_W85M8TFN"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.21.0 - target test 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          25 Mar 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="comment"
    >
      <i
        aria-hidden="true"
        class="tag icon avatar"
      />
      <div
        class="content"
      >
        <a
          class="author text-blue"
          href="undefinedreleases/RE_QQ6FKPXZ"
          rel="noopener noreferrer"
          target="_blank"
        >
          3.15.0 - Release Target in QA 
          <i
            aria-hidden="true"
            class="external small link icon"
          />
        </a>
        <div
          class="metadata"
        >
          on 
          6 Mar 2019
        </div>
        <div
          class="text"
        />
      </div>
    </div>
  </div>
</div>
`;
